src/Team-Services-Bot.Api/Dialogs/ApprovalsDialog.cs

```diff
@@ -0,0 +1,204 @@
+// </summary>
+// ———————————————————————————————
+
+namespace Vsar.TSBot.Dialogs
+{
+    using System;
+    using System.Linq;
+    using System.Runtime.Serialization;
+    using System.Text.RegularExpressions;
+    using System.Threading.Tasks;
+    using System.Web.Http;
+    using Cards;
+    using Microsoft.Bot.Builder.Dialogs;
+    using Microsoft.Bot.Connector;
+    using Microsoft.VisualStudio.Services.ReleaseManagement.WebApi;
+    using Resources;
+
+    /// <summary>
+    /// Represents the dialog to retrieve and approve/reject approvals.
+    /// </summary>
+    [CommandMetadata("approvals")]
+    [Serializable]
+    public class ApprovalsDialog : IDialog<object>
+    {
+        private const string CommandMatchApprovals = "approvals";
+        private const string CommandMatchApprove = @"approve (\d+) *(.*?)$";
+        private const string CommandMatchReject = @"reject (\d+) *(.*?)$";
+
+        [NonSerialized]
+        private IAuthenticationService authenticationService;
+        [NonSerialized]
+        private IVstsService vstsService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApprovalsDialog"/> class.
+        /// </summary>
+        /// <param name="authenticationService">a <see cref="IAuthenticationService"/>.</param>
+        /// <param name="vstsService">The <see cref="IVstsService"/>.</param>
+        public ApprovalsDialog(IAuthenticationService authenticationService, IVstsService vstsService)
+        {
+            this.authenticationService = authenticationService;
+            this.vstsService = vstsService;
+        }
+
+        /// <summary>
+        /// Gets or sets the account.
+        /// </summary>
+        public string Account { get; set; }
+
+        /// <summary>
+        /// Gets or sets the approvalid.
+        /// </summary>
+        public int ApprovalId { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether it is an approval.
+        /// </summary>
+        public bool IsApproval { get; set; }
+
+        /// <summary>
+        /// Gets or sets the profile.
+        /// </summary>
+        public VstsProfile Profile { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Team Project.
+        /// </summary>
+        public string TeamProject { get; set; }
+
+        /// <inheritdoc />
+        public async Task StartAsync(IDialogContext context)
+        {
+            context.Wait(this.ApprovalsAsync);
+
+            await Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Replies with a list of ApprovalCards.
+        /// </summary>
+        /// <param name="context">The <see cref="IDialogContext"/>.</param>
+        /// <param name="result">The <see cref="IAwaitable{T}"/>.</param>
+        /// <returns>An async <see cref="Task"/>/.</returns>
+        public async Task ApprovalsAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
```

Comment:
Got rid of the wrapper, but needed to make the method public, so the unit tests can easily test them.
Also this way the flow is more clear (with the separate method(s))